### DISCUSSION TITLE
Sound output while "Mute Myself" is on - possible bug

### DISCUSSION THREAD

**Discussion by crossrhythm** [jamulussoftware/jamulus#1412] [Bug Investigation]
Hi,
I have been using Jamulus and Zoom to teach violin and chamber music during the pandemic and recently noticed a funny behavior in Jamulus when used with a VAC (Blackhole) and Zoom.  When I press "Mute Myself" in Jamulus, it still sends the sound to Zoom!  I'll be teaching, need to relay something to my wife or kidsin another room, and hit "Mute Myself" so as not to bother my student.  But my audio will still make it's way to Zoom.  It's like the "Mute Myself" only affects the stream out to the Jamulus server, but not between programs on my computer.

Seems like a bug, no?  I would expect "Mute Myself" should really cut local microphone output from Jamulus completely. 

I made a brief demo (while sitting on the couch using wifi - don't worry, I'm usually on ethernet!) here: 
[https://youtu.be/DyfQ4FujVq8](https://youtu.be/DyfQ4FujVq8)

It seems the audio going to Zoom is still the feed coming back from the Jamulus server, complete with gurgle/static.  Does that mean that "Mute Myself" really is just telling the other clients to not play a particular audio stream, but they are all still receiving that stream?  And that locally the VAC is intercepting the sound before it gets to the "Mute Myself" switch for that particular feed?

Workarounds include cutting the gain on my external interface (Scarlett 2i2), or use both "Mute Myself" in Jamulus and Mute in Zoom.

**Comment by hoffie** [comment]
Hi,

> Does that mean that "Mute Myself" really is just telling the other clients to not play a particular audio stream, but they are all still receiving that stream?

No, "Mute myself" sends an empty stream to the server (and therefore, to the other clients). I suspect that Blackhole takes "all audio from Jamulus", which would include the mix from the server (which does not include your sound, but may add the "gurgle" you are describing), but which would also include a Jamulus-generated, local monitoring of your input (which is noticeably different because it does not contain the delay from the client-to-server-to-client roundtrip).

Does that explain things? :)

**Reply by crossrhythm** [reply]
Hmmm... the demo I recorded ([https://youtu.be/DyfQ4FujVq8](https://youtu.be/DyfQ4FujVq8)) seems to contain that distinct gurgle from a less-than-ideal Jamulus connection (I was on wifi).  If it wasn't coming back from the server, why would it have that?  And, with your description, wouldn't you hear yourself twice in Zoom when Mute Myself is off? First the local version, then the one making the round trip to the server?

Not that I don't believe you, just trying to understand it better!

**Reply by ann0see** [reply]
You can verify that it's not coming from the server by cutting internet connection. Maybe the sound issues are from other people?

**Reply by crossrhythm** [reply]
Good idea - I tried it and laughed out loud at the results!  Audio disappeared when I turned off wifi, but then it came back when I clicked "Mute Myself"!  

[https://youtu.be/lPRM2tF9y7w](https://youtu.be/lPRM2tF9y7w)

Which also explains why the audio was bad in the first demo until I clicked "Mute Myself".

**Reply by hoffie** [reply]
> Audio disappeared when I turned off wifi, but then it came back when I clicked "Mute Myself"!

Yes, this may sound confusing and is currently being discussed in #909. :)

**Reply by crossrhythm** [reply]
Yep, that's it.  Thanks, glad it is being discussed.

**Reply by softins** [reply]
> Good idea - I tried it and laughed out loud at the results! Audio disappeared when I turned off wifi, but then it came back when I clicked "Mute Myself"!

If you didn't disconnect from the server within Jamulus, this would be expected. If you turn off wifi while still connected, then obviously, the path to and from the server is interrupted. It may take some time for the client to realise the server is gone. If you then turn on Mute Myself (which admittedly is a confusing name), while Jamulus still thinks it is connected, Jamulus is taking your audio input and mixing it into your audio output, which is why you hear it again.

**Comment by dakhubgit** [comment]
"Mute Myself" mutes yourself to others, not to yourself.  If you want to mute yourself in your own mix, that's what the "M" button in the mixer strip is for.  However, it is not clear whether you are sending your own (headphone) mix to the Zoom session or your own microphone input.  If it's your own microphone input, nothing you can do with Jamulus will change what the Zoom session receives.
